### PR TITLE
Add Munin protocol dissector.

### PR DESCRIPTION
**Pull request by utoni**
* all credits goes to @verzulli

Signed-off-by: Toni Uhlig <matzeton@googlemail.com>

### CODE REVIEW

File: src/lib/protocols/munin.c [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  }
+
+  if (memcmp(packet->payload, munin_prefix, NDPI_STATICSTRING_LEN(munin_prefix)) != 0)
+  {
+    NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+    return;
+  }
+
+  ndpi_int_munin_add_connection(ndpi_struct, flow);
+  flow->host_server_name[0] = '\0';
```
Comment (IvanNardi) [inline src/lib/protocols/munin.c]: Do you think that is really necessary?

File: src/lib/protocols/munin.c [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+  if (packet->payload[packet->payload_packet_len - 1] != '\n')
+  {
+    ndpi_set_risk(ndpi_struct, flow, NDPI_MALFORMED_PACKET, "Missing Munin Hostname");
+    return;
+  }
+
+  size_t host_len = packet->payload_packet_len - NDPI_STATICSTRING_LEN(munin_prefix) - 1;
+  if (host_len > 0)
```
Comment (IvanNardi) [inline src/lib/protocols/munin.c:71]: It seems wrong... if packet->payload_packet_len == NDPI_STATICSTRING_LEN(munin_prefix), the `if` condition is true...
Having `host_len` as a **signed** integer?

Comment (utoni) [reply src/lib/protocols/munin.c:71]: Under those circumstances it should already bail out at `packet->payload[packet->payload_packet_len - 1] != '\n'` or am I wrong?

File: src/lib/protocols/munin.c (hunk 1 above)
Comment (utoni) [reply src/lib/protocols/munin.c]: not really

File: src/lib/protocols/munin.c (hunk 2 above)
Comment (IvanNardi) [reply src/lib/protocols/munin.c:71]: no, you are right. Thanks

**Review by IvanNardi — APPROVED** [review]

**Pull request merged**